### PR TITLE
quickfix.c: cannot specify a count with :chistory, vim-patch:8.1.{1241,1261}, 8.2.{0095,0113}

### LINKED ISSUE #13076: :[count]chistory not supported
- `nvim --version`: 0.4.4
- `vim -u DEFAULTS` (version: 8) behaves differently? yes
- Operating system/version: macOS 10.15.6
- Terminal name/version: terminal.app
- `$TERM`:xterm-256color

neovim supports only simple `:chistory` command. (no `[count]`) 
I can't specify `[count]` parameter. Is this behavior intentional?

Vim8 support `[count]` parameter.

### PR DESCRIPTION
**Pull request by erw7**
Fixes #13076.

#### vim-patch:8.1.1241: Ex command info contains confusing information

Problem:    Ex command info contains confusing information.
Solution:   When using the NOTADR flag use ADDR_OTHER for the address type. Cleanup code using NOTADR.  Check for errors in create_cmdidxs.vim.  Adjust Makefile to see the errors.
vim/vim@b731689

#### vim-patch:8.1.1261: no error for quickfix commands with negative range

Problem:    No error for quickfix commands with negative range.
Solution:   Add ADDR_UNSIGNED and use it for quickfix commands.  Make assert_fails() show the command if the error doesn't match.
vim/vim@25190db

#### vim-patch:8.2.0095: cannot specify exit code for :cquit

Problem:    Cannot specify exit code for :cquit.
Solution:   Add optional argument. (Thinca, Yegappan Lakshmanan, closes vim/vim#5442)
vim/vim@1860bde

#### vim-patch:8.2.0113: "make cmdidxs" fails

Problem:    "make cmdidxs" fails.
Solution:   Allow address for ":cquit".  Add --not-a-term to avoid a delay.
vim/vim@9b24dfc

### CODE REVIEW

File: src/nvim/ex_cmds.lua [hunk 1]
```diff
@@ -483,8 +483,8 @@ return {
   },
   {
     command='chistory',
-    flags=bit.bor(TRLBAR),
-    addr_type=ADDR_LINES,
+    flags=bit.bor(RANGE, COUNT, TRLBAR),
+    addr_type=ADDR_OTHER,
```
Comment (janlazo) [inline src/nvim/ex_cmds.lua:487]: This should be `ADDR_UNSIGNED` which comes from patch v8.1.1261. Is this PR a short term fix for #13076 ?

**Comment by janlazo** [comment]
Patch 8.1.1210, 8.1.1241 should go first before patch 8.1.1261. It's fine if you can make patch 8.1.1261 pass without them.

**Comment by janlazo** [comment]
What's the status of this PR? Any blockers?

**Comment by erw7** [comment]
> What's the status of this PR? Any blockers?

I don't merge changes into the master that have not been approved by anyone, except minor ones. I'm waiting for someone to review it. 

There doesn't seem to be any rules for this project, so maybe it's okay to merge them if they pass the test.

**Comment by janlazo** [comment]
Ok, I'll review.

File: src/nvim/generators/gen_ex_cmds.lua [hunk 2]
```diff
@@ -62,9 +66,24 @@ for _, cmd in ipairs(defs) do
     .cmd_name = (char_u *) "%s",
     .cmd_func = (ex_func_T)&%s,
     .cmd_argt = %uL,
-    .cmd_addr_type = %i
+    .cmd_addr_type = %s
   },
-]], enumname, cmd.command, cmd.func, cmd.flags, cmd.addr_type))
+  ]], enumname, cmd.command, cmd.func, cmd.flags, cmd.addr_type))
```
Comment (janlazo) [inline src/nvim/generators/gen_ex_cmds.lua:71]: Extra whitespace?

File: src/nvim/ex_cmds.lua [hunk 3]
```diff
@@ -15,3288 +15,3277 @@ local COUNT      =    0x400
...
-local RESTRICT   = 0x800000
+local EDITCMD    =   0x4000
+local BUFNAME    =   0x8000
+local BUFUNL     =  0x10000
+local ARGOPT     =  0x20000
+local SBOXOK     =  0x40000
+local CMDWIN     =  0x80000
+local MODIFY     = 0x100000
+local EXFLAGS    = 0x200000
+local RESTRICT   = 0x400000
```
Comment (janlazo) [inline src/nvim/ex_cmds.lua:26]: https://github.com/neovim/neovim/pull/11996 removed restricted mode so `RESTRICT` is now dead code. Maybe comment it out and update affected ex commands.

File: src/nvim/ex_cmds_defs.h [hunk 4]
```diff
@@ -51,32 +51,36 @@
...
+    ADDR_WINDOWS,         // window number
+    ADDR_ARGUMENTS,       // argument number
+    ADDR_LOADED_BUFFERS,  // buffer number of loaded buffer
+    ADDR_BUFFERS,         // buffer number
+    ADDR_TABS,            // tab page number
+    ADDR_TABS_RELATIVE,   // Tab page that only relative
+    ADDR_QUICKFIX_VALID,  // quickfix list valid entry number
+    ADDR_QUICKFIX,        // quickfix list entry number
+    ADDR_UNSIGNED,        // positive count or zero, defaults to 1
+    ADDR_OTHER,           // something else
```
Comment (janlazo) [inline src/nvim/ex_cmds_defs.h:81]: Patch 8.1.1261

`ADDR_OTHER,		 // something else, use line number for '$',`

**Comment by janlazo** [comment]
```
 /*
  * Find an Ex command by its name, either built-in or user.
  * Start of the name can be found at eap->cmd.
- * Returns pointer to char after the command name.
+ * Sets eap->cmdidx and returns a pointer to char after the command name.
  * "full" is set to TRUE if the whole command name matched.
  * Returns NULL for an ambiguous user command.
  */
     static char_u *
 find_command(exarg_T *eap, int *full UNUSED)
```

**Comment by janlazo** [comment]
WIP on reviewing ex_cmds.lua because of the `addr_type` changes.

**Comment by janlazo** [comment]
I ported 8.2.0095. vim/vim#5442 mentions that it is based on Neovim implementation so porting it should not require changes in the C,lua code.

**Pull request closed without merging**